src/services/latex.ts

```
@@ -128,139 +123,117 @@ class Controller_latex extends Controller_keystroke {
-        const count = Math.abs(signedSelectionSize);
-        for (let i = 0; i < count; i += 1) {
-          selectDir(dir);
+      const { restoreInfo } = this.exportLatexSelection(mappedIndices).ctx;
+
+      if (newSelection.startIndex === newSelection.endIndex) {
+        if (restoreInfo?.cursorL) {
+          this.prepareCursorForRestoration().insRightOf(
+            restoreInfo.cursorL as MQNode
+          );
+        } else if (restoreInfo?.cursorParent) {
+          this.prepareCursorForRestoration().insAtLeftEnd(
+            restoreInfo.cursorParent as MQNode
+          );
         }
-      });
-    }
-  }
-
-  // any time there's a selection there is a cursor and anticursor. The
-  // anticursor is the anchor, and the cursor is the head. It should be
-  // true that these are siblings. If you trace right or left far enough
-  // you will reach the other one. This returns the direction and magnitude
-  // of how many hops it takes to find the cursor from the anticursor. Otherwise
-  // returns 0. The idea is to try this both with L and R and use the one, if any,
-  // that comes back with a non-zero answer.
-  private findSignedSelectionSizeInDir(dir: L | R) {
-    const cursor = this.cursor;
-    const anticursor = cursor.anticursor;
-    if (!anticursor) return 0;
-
-    let count = 0;
-    let node = anticursor[dir];
-    while (node !== cursor[dir]) {
-      if (!node) return 0;
-
-      count += dir;
-      node = node[dir];
+      } else {
+        if (restoreInfo?.selectionL && restoreInfo.selectionR) {
+          // TODO - should we validate this selection by verifying you can get from selectionR
+          // to selectionL by traversing leftward? That would be a quick and easy sanity check
+          // to run ahead of time to prevent messing up selection if an invalid selection is
+          // passed in. If an invalid selection is passed in we shouldn't loop infinitely but
+          // we could end up in a weird state.
```

Seems reasonable to add this as a sanity check, to fail loudly and help debugging. But I think it's probably also fine to not do so, since this would really just be a programming error, not something we'd ever expect.

---

I think I'm going to just leave this as a todo for now. I honestly don't know what the best behavior should be if you pass in an invalid selection? Should it error, ignore the restoration, or do it's best? Without a clear cut answer I don't want to write less than ideal error handling.